src/pages/groupComparison/OverlapExclusionIndicator.tsx

```diff
@@ -14,14 +15,32 @@ export default class OverlapExclusionIndicator extends React.Component<IOverlapE
             return null;
         } else {
             const selectionInfo = this.props.store._selectionInfo.result!;
-            if (selectionInfo.overlappingSamples.length === 0 && selectionInfo.overlappingPatients.length === 0) {
+            if ((selectionInfo.overlappingPatients.length === 0 && selectionInfo.overlappingSamples.length === 0) ||
+                (this.props.only === "sample" && selectionInfo.overlappingSamples.length === 0) ||
+                (this.props.only === "patient" && selectionInfo.overlappingPatients.length === 0)) {
                 return null;
             }
 
             let iconClass;
             let alertClass;
             let message;
-            const caseCountsSummary = caseCounts(selectionInfo.overlappingSamples.length, selectionInfo.overlappingPatients.length, " and ");
+            let caseCountsSummary = "";
+            switch (this.props.only) {
+                case "sample":
+                case "patient":
+                    let count = 0;
+                    if (this.props.only === "sample") {
+                        count = selectionInfo.overlappingSamples.length;
+                    } else {
+                        count = selectionInfo.overlappingPatients.length;
+                    }
+                    const plural = (count !== 1);
+                    caseCountsSummary = `${count} ${this.props.only}${plural ? "s" : ""}`;
+                    break;
+                default:
```

Comment:
should there just be a default value for props.only?

Reply:
no, because it's optional and if undefined means its both samples and patients